rfcs/2022-02-24-11552-dd-agent-style-secret-management.md

```diff
@@ -0,0 +1,133 @@
+# RFC <issue#> - 2022-02-xx - Datadog Agent style secret management
+
+The Datadog Agent has a straighforward secret resolution facility to avoid having sensitive information stored directly
+in its config, it relies on a user-provided external program that is run to retrieve sensitive value from a third party
+system. This RFC aims to propose a similar mechanism for Vector.
+
+## Context
+
+The [Datadog Agent documentation][dd-agent-secret-mgmt] provide all user-relevant information to use that feature. It
+covers the specification for the user-provided executable that loads/decrypts secrets along with the agent configuration
+and the syntax to retrieve encrypted config value.
+
+## Cross cutting concerns
+
+- The ongoing [configuration schema work][vector-config-schema-work].
+- `vector config` and related Vector enterprise work.
+
+## Scope
+
+### In scope
+
+- User shall be able to use the same kind of executable to load/decrypt secrets for Vector.
+- This new feature will have a deterministic behaviour when used in conjunction with templates.
+- Situations like topology reload and load/decryption failure will be accounted for.
+
+### Out of scope
+
+- Integration with other secret or distribution configuration like vault, however this RFC will account for that kind of
+  future extension
+
+## Pain
+
+- As of today, secrets like authentication tokens and passwords should be provided inside the topology configuration in
+  plain text and that may not be acceptable in some circumstances.
+- Decoupling secret management and key rotation from configuration management.
+
+## Proposal
+
+### User Experience
```

Comment:
Also, I think we should handle reloading the secrets on SIGHUP as users generally expect Vector to reload all of its configuration. We may want to call that out explicitly here to make sure we don't forget to cover it.